etcdmain/grpc_proxy.go

```diff
@@ -127,12 +132,23 @@ func newGRPCProxyStartCommand() *cobra.Command {
 	// experimental flags
 	cmd.Flags().BoolVar(&grpcProxyEnableOrdering, "experimental-serializable-ordering", false, "Ensure serializable reads have monotonically increasing store revisions across endpoints.")
 	cmd.Flags().StringVar(&grpcProxyLeasing, "experimental-leasing-prefix", "", "leasing metadata prefix for disconnected linearized reads.")
+
+	cmd.Flags().BoolVar(&grpcProxyDebug, "debug", false, "Enable debug-level logging for grpc-proxy.")
+
 	return &cmd
 }
 
 func startGRPCProxy(cmd *cobra.Command, args []string) {
 	checkArgs()
 
+	capnslog.SetGlobalLogLevel(capnslog.INFO)
+	if grpcProxyDebug {
+		capnslog.SetGlobalLogLevel(capnslog.DEBUG)
+		grpc.EnableTracing = true
+	} else {
+		grpclog.SetLoggerV2(grpclog.NewLoggerV2(ioutil.Discard, ioutil.Discard, ioutil.Discard))
```

Comment:
well, we should still return gRPC loggings whose level >= Error.

Reply:
How about `NewLoggerV2WithVerbosity` to stderr with verbosity 2?

https://github.com/grpc/grpc-go/blob/ec61302c8eeed1f9705bc51e6606220ce21f4a60/transport/log.go#L26

Do we want to enable these as well?

https://github.com/coreos/etcd/blob/a7f1fbe00ec216fcb3a1919397a103b41dca8413/embed/config.go#L264

https://github.com/coreos/etcd/blob/a7f1fbe00ec216fcb3a1919397a103b41dca8413/clientv3/logger.go#L40

Reply:
@xiang90 Turns out grpc uses leveled log only in the [`transport`](https://github.com/grpc/grpc-go/blob/ec61302c8eeed1f9705bc51e6606220ce21f4a60/transport/log.go#L26) package. Everywhere else logs with `grpclog.Infof/Warningf/...`, which means we need either disable them all or print them all including verbose internal balancer logs. There's no way to print only errors.